wiki_articles/compilation_steps.md

```diff
@@ -0,0 +1,26 @@
+# How Is C/C++ Source Code Compiled?
+
+\:hash: The **preprocessor** expands macros and copy-pastes the content of included files
+at their point of inclusion. After being preprocessed, a source file no longer
+contains preprocessor directives. The result is called a _translation unit_ (TU).
+
+> :bulb: _The preprocessor merely does text replacement, it does not care about any aspect of C/C++._
+
+:ce: The **compiler** turns instructions into machine code and enforces C/C++
```

Comment:
Oh and one last thing, at the moment supported emojis is a somewhat manual substitution (there's a `substitute_emojis` function in the parser that handles 4 supported emojis). It's probably best if I try to update this handling separately to be more robust.

Reply:
I looked around and did what seems to have been done for the `:stackoverflow:` emoji in the value_categories.md article, it gives `<:ce:1211139919884582943>`. Discord seems to be okay with that: https://discord.com/channels/331718482485837825/506274405500977153/1220681818509021205

Reply:
That'll work!